source/bsd/kqueue_event_loop.c

```diff
@@ -294,6 +297,7 @@ static void s_destroy(struct aws_event_loop *event_loop) {
         assert("Failed to destroy event-thread, resources have been leaked." == NULL);
         return;
     }
+    aws_atomic_store_int(&impl->thread_id, (size_t)aws_thread_current_thread_id());
```

Comment:
I would really like to see a comment at the sites where thread ids are transitioned to explain why